pkg/services/cleanup/cleanup.go

```diff
@@ -37,9 +38,14 @@ func (srv *CleanUpService) Run(ctx context.Context) error {
 	for {
 		select {
 		case <-ticker.C:
+			ctxWithTimeout, cancelFn := context.WithTimeout(ctx, time.Minute*9)
+			defer cancelFn()
+
 			srv.cleanUpTmpFiles()
 			srv.deleteExpiredSnapshots()
 			srv.deleteExpiredDashboardVersions()
+			srv.cleanUpOldAnnotations(ctxWithTimeout)
```

Comment:
Shouldn't this, snapshots, dashboard versions be wrapped in `ServerLockService.LockAndExecute`?

Reply:
It's fine running this cleanup job in parallel. If multiple servers start racing, one server might not be able to delete any annotations, which in case it will just stop and other servers can keep deleting.

Reply:
Okay if you say so